rule bwa_index:
    input:
        "genome/{reference}.fa"
    output:
        "genome/{reference}.fa.amb",
        "genome/{reference}.fa.ann",
        "genome/{reference}.fa.bwt",
        "genome/{reference}.fa.pac",
        "genome/{reference}.fa.sa"
    shell:
        "bwa index {input}"

rule bwa_align:
    input:
        "genome/{reference}.fa.amb",
        "genome/{reference}.fa.ann",
        "genome/{reference}.fa.bwt",
        "genome/{reference}.fa.pac",
        "genome/{reference}.fa.sa",
        ref="genome/{reference}.fa",
        reads="test_samples/BGI/{sample}.fastq.gz"
    log:
        "logs/bwa.{reference}.{sample}.log"
    output:
        "{reference}.{sample}.unsorted.bam"
    shell:
        "bwa mem {input.ref} {input.reads} 2>{log} | samtools view -b > {output}"

rule bam_sort:
    input:
        rules.bwa_align.output
    output:
        "{reference}.{sample}.sorted.bam"
    shell:
        "samtools sort {input} -o {output}"

rule get_unmapped_reads:
    input:
        rules.bam_sort.output
    output:
        "{reference}.{sample}.unmapped.bam"
    shell:
        "samtools view -f4 {input} > {output}"